src/lib/sql/sql_pipeline_statement.hpp

```diff
@@ -85,6 +92,9 @@ class SQLPipelineStatement : public Noncopyable {
   const bool _use_mvcc;
   const bool _auto_commit;
   std::shared_ptr<TransactionContext> _transaction_context;
+
+  // Caching
+  static SQLQueryCache<SQLQueryPlan> _query_plan_cache;
```

Comment:
use a meyer's singleton and probably make the function free - imo the cache shouldn't be owned by SQLPipelineStatement

Reply:
I'm not sure I like that better. Where would the function be located? For it to fit into the common Singleton::get() pattern that we have so far, we would need SQLQueryCache<SQLQueryPlan>::get() or something like that which looks weird. And having a global free function get_query_plan_cache() seems wrong too. Or did you have a different solution in mind?

Reply:
meyer's singleton - a must given this is concurrently accessed
the name and scope of the function which contains the singleton - up to debate, but as you mentioned `SQLQueryCache<SQLQueryPlan>::get()` would be consistent with the rest of the codebase. I'd go for it.

(And imo the reason free functions feel "wrong" is only the year-long OO-indoctrination we've been exposed to ;) )